client/state/sites/reducer.js

```diff
@@ -204,6 +205,17 @@ export function items( state = {}, action ) {
 			}, state );
 		}
 
+		case SITE_DOMAINS_PRIMARY_SET_COMPLETED: {
+			const { siteId, primaryDomainName } = action;
+			return {
+				...state,
+				[ siteId ]: {
+					...state[ siteId ],
+					URL: `http://${ primaryDomainName }`
```

Comment:
@aidvu @klimeryk @umurkontaci Any thoughts on this one?

I'm on support rotation this week. If you have free bandwidth this week, feel free to take over this PR.